modules/services/linux-wallpaperengine.nix

```
@@ -0,0 +1,121 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.linux-wallpaperengine;
+
+in {
+  meta.maintainers = [ hm.maintainers.ckgxrg ];
+
+  options.services.linux-wallpaperengine = {
+    enable = mkEnableOption
+      "linux-wallpaperengine, an implementation of Wallpaper Engine functionality";
+
+    package = mkPackageOption pkgs "linux-wallpaperengine" { };
+
+    assetsPath = mkOption {
+      type = types.path;
+      description = "Path to the assets directory.";
+    };
+
+    clamping = mkOption {
+      type = types.nullOr (types.enum [ "clamp" "border" "repeat" ]);
+      default = null;
+      description = "Clamping mode for all wallpapers.";
+    };
+
+    wallpapers = mkOption {
+      type = types.listOf (types.submodule {
+        options = {
+          monitor = mkOption {
+            type = types.str;
+            description = "Which monitor to display the wallpaper.";
+          };
+
+          wallpaperId = mkOption {
+            type = types.str;
+            description = "Wallpaper ID to be used.";
+          };
+
+          extraOptions = mkOption {
+            type = types.listOf types.str;
+            default = [ ];
+            description =
+              "Extra arguments to pass to the linux-wallpaperengine command for this wallpaper.";
+          };
+
+          scaling = mkOption {
+            type =
+              types.nullOr (types.enum [ "stretch" "fit" "fill" "default" ]);
+            default = null;
+            description = "Scaling mode for this wallpaper.";
+          };
+
+          fps = mkOption {
+            type = types.nullOr types.int;
+            default = null;
+            description = "Limits the FPS to a given number.";
+          };
+
+          audio = {
+            silent = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Mutes all sound of the wallpaper.";
+            };
+
+            automute = mkOption {
+              type = types.bool;
+              default = true;
+              description = "Automute when another app is playing sound.";
+            };
+
+            audio-processing = mkOption {
```

Perhaps can skip the "audio-" part since it's already inside the audio attribute set?

```suggestion
            processing = mkOption {
```

---

I did this according to the original project's cmdline `--no-audio-processing`.
Anyways it looks good to me, I'll change it